torch/testing/_internal/common_utils.py

```diff
@@ -2337,6 +2337,13 @@ def to_list(input):
         elif isinstance(x, Sequence) and isinstance(y, torch.Tensor):
             x = torch.as_tensor(x, dtype=y.dtype, device=y.device)
 
+        # If x or y are tensors and nested then we unbind them to a list of tensors this should allow us to compare
```

Comment:
I'm not sure that's always true. Especially because of autograd etc. etc.

Checkout the code in [pytorch/nestedtensor](https://github.com/pytorch/nestedtensor/blob/master/test/utils_test_case.py#L232-L283).

Reply:
Do you think it would be easier/better to create a new derived class for NestedTestCase that has this override? Defined in test/test_nesetedtensor.py ?

Reply:
I'd personally keep it either here or in a helper function that's called into when either input is found to be a nested tensor.

Reply:
Okay, I am not totally sure what differences could be slipping through by unbinding them then checking equality. AFAIK this matches the previous  nt_equal that was defined in test/test_nestedtensor. The dtype and device get checked by the base assertEqual. Is there other metadata specific to NT that gets removed by running unbind that needs to be `and` with the base assertEqual on the unbinded tensors.

Reply:
@jbschlosser re-pinging to make sure this doesn't turn into a zombie

Reply:
@drisspg Thanks for the re-ping :)

> Okay, I am not totally sure what differences could be slipping through by unbinding them then checking equality.

What about `requires_grad`, stride metadata, etc.? I guess if `unbind()` is implemented correctly, it inherits those correctly from the `NestedTensorImpl`. Now that we have `NestedTensorImpl` backed by a storage instead of a `Tensor`, I think this removes some possible discrepancies between the metadata for the `NestedTensorImpl` and its underlying components.

I'm okay with saying two NTs are equal if their unbind results are "equal" (`allclose()`).